server/interceptor/interceptor.go

```diff
@@ -40,6 +41,11 @@ const (
 )
 
 var (
+	// PutTokensTimeout is the timeout used for PutTokens calls.
+	// PutTokens happens in a separate goroutine and with an independent context, therefore it has
+	// its own timeout, separate from the RPC that causes the calls.
+	PutTokensTimeout = 5 * time.Second
```

Comment:
Should this be a flag?

Reply:
It could be a flag, that's why it's setup this way, but I don't think it warrants a flag right now. We can put one in rather quickly if we need it.